agent/api/task/task.go

```diff
@@ -2338,59 +2338,147 @@ func (task *Task) dockerLinks(container *apicontainer.Container, dockerContainer
 			if err != nil {
 				return nil, err
 			}
+
+			scContainer := task.GetServiceConnectContainer()
 			if taskContainer == scContainer {
-				// create bindings for all ingress listener ports
-				// no need to create binding for egress listener port as it won't be access from host level or from outside
-				for _, ic := range task.ServiceConnectConfig.IngressConfig {
-					listenerPortInt := int(ic.ListenerPort)
-					dockerPort := nat.Port(strconv.Itoa(listenerPortInt)) + "/tcp"
-					hostPort := 0           // default bridge-mode SC experience - host port will be an ephemeral port assigned by docker
-					if ic.HostPort != nil { // non-default bridge-mode SC experience - host port specified by customer
-						hostPort = int(*ic.HostPort)
-					}
-					dockerPortMap[dockerPort] = append(dockerPortMap[dockerPort], nat.PortBinding{HostPort: strconv.Itoa(hostPort)})
-					// append non-range, singular container port to the containerPortSet
-					containerPortSet[listenerPortInt] = struct{}{}
-					// set taskContainer.ContainerPortSet to be used during network binding creation
-					taskContainer.SetContainerPortSet(containerPortSet)
+				// If the associated task container is the service connect AppNet container,
+				// create port binding(s) for ingress listener ports based on its ingress config.
+				// Note that, there is no need to do this for egress listener port as it won't be accessed
+				// from host level or from outside.
+				dockerPortMap, containerPortSet, err := task.buildPortMapWithSCIngressConfig(dynamicHostPortRange)
+				if err != nil {
+					logger.Error("Failed to build a port map with service connect ingress config", logger.Fields{
+						field.TaskID:           task.GetID(),
+						field.Container:        taskContainer.Name,
+						"dynamicHostPortRange": dynamicHostPortRange,
+						field.Error:            err,
+					})
+					return nil, err
 				}
+				// Set taskContainer.ContainerPortSet to be used during network binding creation
+				taskContainer.SetContainerPortSet(containerPortSet)
 				return dockerPortMap, nil
 			}
 			containerToCheck = taskContainer
 		} else {
-			// If container is neither SC container nor pause container, it's a regular task container. Its port bindings(s)
-			// are published by the associated pause container, and we leave the map empty here (docker would actually complain
-			// otherwise).
+			// If the container is neither service connect AppNet container nor pause container, and it is a regular task container.
+			// Its port bindings(s) are published by the associated pause container, and we will leave the map empty here
+			// (docker would actually complain otherwise).
+			//
+			// Note - for service connect bridge mode, we do not allow customers to specify a host port for their application containers.
+			// Additionally, AppNet will NOT proxy traffic to that port when an ephemeral host port is assigned.
 			return dockerPortMap, nil
```

Comment:
nit: maybe we could add a debug message here saying it's a regular task container and we're leaving the map empty  (I also really like these comments as well)

Reply:
Thanks for the suggestion! I am not sure if adding a debug log here helps because  
1. Customers don't really have to know what happened to application containers and service connect AppNet containers as port bindings are published by the associated pause containers.
2. Customers can check network bindings results for service connect AppNet container and application containers through [describe-tasks]((https://docs.aws.amazon.com/cli/latest/reference/ecs/describe-tasks.html)
3. Showing an empty map in logs probably has less value to customers and us to debug